src/declarations/memory.d.ts

```diff
@@ -1,4 +1,11 @@
 type operationMode = 'manual' | 'semiautomatic' | 'automatic';
+/**
+ * 0: Basic
+ * 1: Collect from enemy storage/terminal
+ * 2: Collect from all sources TBD
+ * 3: Collect all and mine walls for energy TBD
+ */
+type resourceCollectionMode = number;
```

Comment:
why do you think this needs its own setting?

Reply:
I was thinking of setting some general principles for resource collection. Taking from all structures for example, and even mining walls.

As a more general logic, an increased resource collection could represent things like spawning haulers to pick up energy from SK room creep drops, collecting from invaders, general principles for mining, etc.
With how often overminds get wiped, starting from a blank slate you're desperate for any and all energy you can get a hold of. 
I had a grander vision for this but then realized it was out of scope, happy to remove from being a constant for now.